### PR TITLE
feat: migrate `POST /api/workspaces/:workspace_id/users` to API v1

### LINKED ISSUE #155: [TASK] Migrate `POST /api/workspaces/:workspace_id/users/:user_id` endpoint

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR add a new endpoint `POST /api/v1/workspaces/:workspace_id/users` to create a new user for a workspace.

Changes introduced in this new endpoint compared to the old one:
* Now we must provide the `user_id` in the body of the `POST` request instead of adding it in the URL.
* I have changed the `create_workspace_user` context function so we need an additional change in `argilla` repo. The PR with the changes on argilla is here: https://github.com/argilla-io/argilla/pull/4807
* As done with previous PRs the returned user doesn't has `full_name` or `workspaces`.

Closes #155 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

- [x] Adding new tests for the endpoint.

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla_server/apis/v1/handlers/workspaces.py [hunk 1]
```diff
@@ -131,3 +131,39 @@ async def list_workspace_users(
...
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"User with id `{workspace_user_create.user_id}` not found",
+        )
+
+    workspace_user = await accounts.get_workspace_user_by_workspace_id_and_user_id(db, workspace_id, user.id)
+    if workspace_user is not None:
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail=f"User with id `{user.id}` already exists in workspace with id `{workspace_id}`",
+        )
```
Comment (frascuchon) [inline src/argilla_server/apis/v1/handlers/workspaces.py:165]: Not sure if this is a context responsibility

File: src/argilla_server/apis/v1/handlers/workspaces.py [hunk 2]
```diff
@@ -131,3 +131,39 @@ async def list_workspace_users(
...
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Workspace with id `{workspace_id}` not found",
+        )
+
+    user = await accounts.get_user_by_id(db, workspace_user_create.user_id)
+    if user is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"User with id `{workspace_user_create.user_id}` not found",
+        )
```
Comment (frascuchon) [inline src/argilla_server/apis/v1/handlers/workspaces.py:158]: For other endpoints where an id is passed as a body param, we're returning a 422 in this case. I think it makes sense.

File: src/argilla_server/apis/v1/handlers/workspaces.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla_server/apis/v1/handlers/workspaces.py:165]: It is, I will move it to the context.

File: src/argilla_server/apis/v1/handlers/workspaces.py (hunk 2 above)
Comment (jfcalvo) [reply src/argilla_server/apis/v1/handlers/workspaces.py:158]: Ok, I will take a look.

File: src/argilla_server/apis/v1/handlers/workspaces.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla_server/apis/v1/handlers/workspaces.py]: Done.

File: src/argilla_server/apis/v1/handlers/workspaces.py (hunk 2 above)
Comment (jfcalvo) [reply src/argilla_server/apis/v1/handlers/workspaces.py]: Done.

File: src/argilla_server/errors/future/base_errors.py [hunk 3]
```diff
@@ -21,6 +21,12 @@ class NotFoundError(Exception):
     pass
 
 
+class NotUniqueError(Exception):
```
Comment (frascuchon) [inline src/argilla_server/errors/future/base_errors.py:24]: +1

**Review by frascuchon — APPROVED** [review]

**Pull request merged**